### ISSUE TITLE
Instabilities, freezes and springboard crashes on unc0ver ios12. 

### ISSUE THREAD

**Issue by GioSayah** [Matchstic/Xen-HTML#83]
‪ hey man, i appreciate all your work, but I’ve been experiencing major instabilities while using xenhtml, from random freezes and crashes, sometimes even a random reboot. In addition i noticed than whenever i have xenhtml installed i cannot jailbreak using unc0ver with "reload system daemons" enabled, whenever i have xen installed i go through 9-10 Sb crashes per day, I've had it disabled for 2 days now without a single crash or freeze and now unc0ver works flawlessly whenever i had to reboot the device. About the crashes as i said, it's completely random, as some has mentioned when music or media is playing the crashes happen more often but it's not limited to that, sometimes while invoking the notification center the phone resprings and other just while scrolling in a random app, whatsapp ig or even settings. Deleted xen for 2 days and none of the above occurred again. I can help debugging whenever it's needed.

**Comment by Matchstic** [comment]
Have you checked that it’s not XenInfo causing these issues?

**Comment by GioSayah** [comment]
Yes for my last day of pinpointing my issues i had xeninfo deleted, and i was using a widget that does not have xeninfo as a dependency. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by GioSayah** [comment]
@matchstic Yes for my during my last day of pinpointing the issues i hd xeninfo deleted, and i was using a widget that does not have xeninfo as a dependency.

**Comment by Matchstic** [comment]
Hmm. I have not experienced this at all myself. What device is this on, what jailbreak is used, and which iOS version?

I am very reluctant to accept issues like this for Xen HTML, because usually issues like this tend to come from XenInfo, not this tweak.

**Comment by GioSayah** [comment]
@matchstic i understand where you coming from, but neither i would have posted this here before i was absolutely sure that i pinpointed the problem precisely. I kept testing things out for about 4 days before i narrowed it down to xenhtml. I'm using iphone X ios 12.0 uncover beta 41, and i can 100% confirm that the issue about "reloading daemons" and the jailbreak process respring looping only when xen html is installed.

**Comment by Matchstic** [comment]
Okay. Usually, I get people just assuming it’s my fault without doing any proper checks.

For the respring loop, do you get any crash reports generated for SpringBoard or backboardd? I’m not able to check myself, as my device (pretty much the same configuration, on iOS 12.1.1 instead) does not exhibit these issues at all

**Comment by GioSayah** [comment]
@matchstic as mentioned earlier the respring "loop" only happens when xenhtml is installed and I'm trying to jailbreak my device from an unjailbroken state (while having "reload system daemons" enabled) 
The crashes and resprings that happen randomly are usually a 5 secs respring loading screen that back to lockscreen (still in jailbroken state). About the logs i have crash reporter and cr4shed install and i have a couple of Screenshots of the kind of logs i was getting. But yes some of them were for springboard, i don't recall seeing anything under backboard though.

**Comment by Matchstic** [comment]
Excellent, logs will really help here! Please can you cause the respring loop again, and send over whatever is generated for SpringBoard. A good way to do that is to copy/paste the contents of the log to ghostbin.com and link it here. A screenshot doesn’t help me that much unfortunately.

**Comment by GioSayah** [comment]
@matchstic it's kinda late now, so i'll probably get it done by tomorrow and send the logs hoping to fix this issue.

**Comment by Matchstic** [comment]
Thanks, I’m in no rush; whenever is fine

**Comment by JunesiPhone** [comment]
I would like to suggest (only as this wasn't stated) removing/disabling all other tweaks besides XenHTML, this would mean keep XenHTML as the only tweak enabled, with it's widgets disabled and seeing if the issue persists. One of my devices is an iPX 12.1 and i've had the looping before, but it was never XenHTML. I'm not saying this is the same for you as weird things have been reported on this jailbreak, but this would rule out a conflict. Maybe XenHTML isn't the issue, but when it's installed with another tweak it conflicts and causes this issue.

Another note, I've never used "reload system daemons" as i've never needed to. All that was needed with the loop was to disable tweaks, then delete one at a time until the loop stopped. I then disabled all other tweaks but that specific one and seen if it happened. For the multiple times I did this my issue ended up being CrashReporter and Cr4shed (note this was before rocket bootstrap was updated and an earlier version of the jailbreak) I see you also have CrashReporter and Cr4shed installed.
